vsphere_plugin_common/__init__.py

```diff
@@ -1565,7 +1566,18 @@ def create_server(
         datacenter = self._get_obj_by_name(vim.Datacenter,
                                            datacenter_name)
 
-        destfolder = datacenter.vmFolder
+        if not vm_folder:
+            destfolder = datacenter.vmFolder
+        else:
+            folder = self._get_obj_by_name(vim.Folder, vm_folder)
+            if not folder:
+                raise NonRecoverableError(
+                    'Could not use existing vm_folder "{name}" as no '
```

Comment:
I don't think you need the word 'existing' here, it's implied by stating that it doesn't exist.